packages/react/src/primitives/shared/utils.ts

```diff
@@ -159,3 +159,15 @@ export const mergeVariantsAndOverrides = (
     ...merged,
   };
 };
+
+export type Modifiers = string | number | null;
```

Comment:
👍🏽 Still curious, can we narrow the allowed values of `Modifiers`? Also, think it can be the singular form `Modifier` since it is not an array or an object